backend/src/data/repositories/tag.repository.ts

```diff
@@ -7,6 +7,10 @@ class TagRepository extends Repository<Tag> {
     return this.findOne({ id });
   }
 
+  public findByNameAndWorkspaceId(workspaceId: string, name: string): Promise<Tag[]> {
+    return this.find({ workspaceId, name });
+  }
+
   public findAllByWorkspaceId(workspaceId: string): Promise<Tag[]> {
```

Comment:
Just remove `.addSelect('tag.workspaceId')` here and you will return Promise<ITag[]>, not need for mapping

```suggestion
  public findAllByWorkspaceId(workspaceId: string): Promise<ITag[]> {
    return this.createQueryBuilder('tag')
      .select('tag.id')
      .addSelect('tag.name')
      .where('tag.workspaceId= :workspaceId', { workspaceId })
      .orderBy('tag.name', 'ASC')
      .getMany();
  }
```